test/Microsoft.Azure.WebJobs.Extensions.Kafka.EndToEndTests/KafkaEndToEndTests.cs

```diff
@@ -265,14 +266,33 @@ public async Task Multiple_Hosts_Process_Events_At_Least_Once()
             var producerHost = await this.StartHostAsync(typeof(KafkaOutputFunctions));
             var producerJobHost = producerHost.GetJobHost();
 
-            var producerTask = producerJobHost.CallOutputTriggerStringAsync(
-                GetStaticMethod(typeof(KafkaOutputFunctions), nameof(KafkaOutputFunctions.SendToStringTopic)),
-                this.endToEndTestFixture.StringTopicWithTenPartitions.Name,
-                Enumerable.Range(1, producedMessagesCount).Select(x => EndToEndTestExtensions.CreateMessageValue(messagePrefix, x)), 
-                TimeSpan.FromMilliseconds(100));
+            var host2HasPartitionsSemaphore = new SemaphoreSlim(0);
 
-            IHost host1 = null, host2 = null;
+            // Split the call in 4, waiting 1sec between calls
+            var producerTask = Task.Run(async () => 
+            {
+                var allMessages = Enumerable.Range(1, producedMessagesCount).Select(x => EndToEndTestExtensions.CreateMessageValue(messagePrefix, x));
+                const int loopCount = 4;
+                var itemsPerLoop = producedMessagesCount / loopCount;
+                for (var i=0; i < loopCount; ++i)
+                {
+                    var messages = allMessages.Skip(i * itemsPerLoop).Take(itemsPerLoop);
+                    await producerJobHost.CallOutputTriggerStringAsync(
+                        GetStaticMethod(typeof(KafkaOutputFunctions), nameof(KafkaOutputFunctions.SendToStringTopic)),
+                        this.endToEndTestFixture.StringTopicWithTenPartitions.Name,
+                        messages);
+
+                    if (i == 0)
```

Comment:
Instead of doing this check here, could you just move the entire execution of this block of code (`producerTask`) to an async/await method call down on line 329? Are we _really_ gaining anything (esp since this is only a test) by kicking off this `Task.Run`, but effectively waiting for line 322 to complete before doing any real work (beyond the first chunk)? Would remove the need for the semaphore, etc.

Reply:
I can't. This test does the following:
- In a separated task producer creates 4x80 items. After the first batch is created waits for semaphore.
- In main task host1 starts processing messages
- When host1 has at least a message starts hosts2
- When host2 obtains at least 1 partitions it triggers the semaphore
- Once the producer tasks is finished (all 240 messages were created), validate that all messages were processed by host1 and host2